catalyst-gateway/bin/src/event_db/follower.rs

```diff
@@ -46,68 +48,131 @@ impl EventDB {
         Ok(())
     }
 
+    /// Get current slot info for the provided date-time and network
+    pub(crate) async fn current_slot_info(
+        &self, date_time: DateTime, network: Network,
+    ) -> Result<(SlotNumber, BlockHash, DateTime), Error> {
+        let conn = self.pool.get().await?;
+
+        let rows = conn
+            .query(
+                include_str!(
+                    "../../../event-db/queries/follower/select_slot_index_by_datetime/current_slot_info.sql"
```

Comment:
This is not very DRY.
And i think it shows a big weakness in our use of external SQL files.
The only difference between these three functions is a single comparison operator inside the query.
You can't parameterize the comparison operator in a query, but it would be trivial if we were building the query here inline.

This should be a single function,  takes a "current/previous/next" argument, and builds a query that reflects it.
Not three functions that are identical except for the sql template, which themselves are identical except for the comparison operator.

Reply:
done as we discussed